config/opal_config_pmix.m4

```diff
@@ -0,0 +1,200 @@
+dnl         internally, to the header file that should be included for
+dnl         embedded builds.  This is used by PRRTE, but should not
+dnl         be used by new code.
+dnl   * opal_pmix_mode - either external or internal.  If internal,
+dnl         --with-pmix should be ignored by other packages
+dnl   * opal_pmix_CPPFLAGS - the C Preprocessor flags necessary to
+dnl         run the preprocessor on a file which relies on PMIx
+dnl         headers.  This will be folded into the global CPPFLAGS,
+dnl         so most people should never need this.
+dnl   * opal_pmix_LDFLAGS - the linker flags necessary to run the
+dnl         linker on a file which relies on PMIx libraries.  This
+dnl         will be folded into the global CPPFLAGS, so most people
+dnl         should never need this.
+dnl   * opal_pmix_LIBS - the libraries necessary to link source which
+dnl         uses PMIx.  Cannot be added to LIBS yet, because then
+dnl         other execution tests later in configure (there are sadly
+dnl         some) would fail if the path in LDFLAGS was not added to
+dnl         LD_LIBRARY_PATH.
+dnl   * CPPFLAGS, LDFLAGS - Updated opal_pmix_CPPFLAGS and
+dnl         opal_pmix_LDFLAGS.
+AC_DEFUN([OPAL_CONFIG_PMIX], [
+    OPAL_VAR_SCOPE_PUSH([external_pmix_happy internal_pmix_happy internal_pmix_args internal_pmix_libs])
+
+    opal_show_subtitle "Configuring PMIx"
+
+    OPAL_3RDPARTY_WITH([pmix], [pmix], [package_pmix])
+
+    opal_pmix_header=""
+
+    internal_pmix_happy=0
+    m4_ifdef([package_pmix],
+        [# always configure the internal pmix, so that
+         # make dist always works.
+	 internal_pmix_args="--without-tests-examples --disable-pmix-binaries --disable-pmix-backward-compatibility --disable-visibility"
+         internal_pmix_libs=
+
+         AS_IF([test "$opal_libevent_mode" = "internal"],
+               [internal_pmix_args="$internal_pmix_args --with-libevent=cobuild"
+                internal_pmix_libs="$internal_pmix_libs $opal_libevent_LIBS"])
+
+         AS_IF([test "$opal_hwloc_mode" = "internal"],
+               [internal_pmix_args="$internal_pmix_args --with-hwloc=cobuild"
+                internal_pmix_libs="$internal_pmix_libs $opal_hwloc_LIBS"])
+
+         AS_IF([test ! -z "$internal_pmix_libs"],
+               [internal_pmix_args="$internal_pmix_args --with-prte-extra-lib=\"$internal_pmix_libs\""])
+
+         # Pass all our compiler/linker flags to PMIx, so that it
+         # picks up how to build an internal HWLOC and libevent, plus
+         # picks up any user-specified compiler flags from the master
+         # configure run.
+         export CFLAGS CPPFLAGS LDFLAGS
+         PAC_CONFIG_SUBDIR_ARGS([3rd-party/openpmix], [$internal_pmix_args],
+                           [[--with-libevent=internal], [--with-hwloc=internal],
+                            [--with-libevent=external], [--with-hwloc=external],
+                            [--with-pmix=.*], [--with-platform=.*]],
```

Comment:
PMIx is now looking for `--with-pmix-platform`

Reply:
Yup; we don't actually have to do anything here.  The 3rd argument is an array of options that PAC_CONFIG_SUBDIR_ARGS will strip out from the top-level configure options when passing down to the sub-configure.  This will avoid passing the `--with-platform` argument to PMIx, as belt and suspenders type thing.